jsonschema-core.xml

```diff
@@ -2337,15 +2339,18 @@
                         </t>
                         <t>
                             An array instance is valid against "contains" if at least one of
-                            its elements is valid against the given schema. The subschema MUST be
-                            applied to every array element even after the first match has
-                            been found, in order to collect annotations for use by other keywords.
-                            This is to ensure that all possible annotations are collected.
+                            its elements is valid against the given schema, provided that
+                            "minContains" is not present or is a value greater than 0.
                         </t>
                         <t>
-                            Logically, the validation result of applying the value subschema to each
-                            item in the array MUST be ORed with "false", resulting in an overall
-                            validation result.
+                            If "minContains" is present and has a value of 0, then an array instance
+                            is valid against this keyword only if NONE of its elements is valid
+                            against the given schema.
```

Comment:
I think this is backwards. If `"minContains": 0` then `contains` should always valid no matter what items are in the array.

Reply:
I would agree. If `minContains` were 1, it would be a no-op. If `minContains` were 2, at least 2 of the items must match `contains`. If you say "At least N of the items in the array must match", then when N is 0, the logical reading is 0 of the items must match `contains`.

For none of the items, you would have to do `not > contains`.

Why would anyone want to do N = 0?
It would be an odd use case seening as we don't see many applications that use annotations at current, but it could be a means to add annotations to items in an array without causing validation assertions if there are none that match the `contains` schema value.

Reply:
In fact, later we say...

> A value of 0 causes "contains" to always pass validation (but validation can still fail against a "maxContains" keyword).

In the validation spec doc.

Reply:
> I think this is backwards

Yes, I agree. I don't remember why I put this in here; `contains` should behave the same way no matter what `minContains` and `maxContains` say. Any additional constraints added by those keywords should result in errors being generated from those keywords, not `contains` itself.